src/test/java/org/jenkinsci/plugins/runselector/filters/ParameterizedRunFilterTest.java

```diff
@@ -0,0 +1,252 @@
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.JenkinsRule.WebClient;
+
+import hudson.model.Item;
+import hudson.model.ParameterDefinition;
+import hudson.model.ParametersAction;
+import hudson.model.ParametersDefinitionProperty;
+import hudson.model.Result;
+import hudson.model.StringParameterDefinition;
+import hudson.model.StringParameterValue;
+
+/**
+ * Tests for {@link RunFilterParameter} and {@link ParameterizedRunFilter}
+ */
+public class ParameterizedRunFilterTest {
+    @ClassRule
+    public static final JenkinsRule j = new JenkinsRule();
+
+    private static WorkflowJob jobToCopyFrom;
+    private static WorkflowRun runToCopyFrom1;
+    private static WorkflowRun runToCopyFrom2;
+
+    @BeforeClass
+    public static void prepareBuildsToCopyFrom() throws Exception {
+        jobToCopyFrom = j.jenkins.createProject(
+            WorkflowJob.class,
+            RandomStringUtils.randomAlphanumeric(7)
+        );
+        jobToCopyFrom.setDefinition(new CpsFlowDefinition("// do nothing"));
+        runToCopyFrom1 = j.assertBuildStatusSuccess(jobToCopyFrom.scheduleBuild2(0));
+        runToCopyFrom2 = j.assertBuildStatusSuccess(jobToCopyFrom.scheduleBuild2(0));
+        runToCopyFrom1.keepLog();
+    }
+
+    @Test
+    public void testConfigureBuildFilterParameter() throws Exception {
+        RunFilterParameter param = new RunFilterParameter(
+            "PARAM",
+            "description",
+            new AndRunFilter(
+                new ParametersRunFilter("PARAM1=VALUE1"),
+                new SavedRunFilter()
+            )
+        );
+        WorkflowJob job = j.jenkins.createProject(
+                WorkflowJob.class,
+                RandomStringUtils.randomAlphanumeric(7)
+        );
+        job.addProperty(new ParametersDefinitionProperty(
+            Arrays.<ParameterDefinition>asList(param)
+        ));
+        j.configRoundtrip((Item)job);
+        j.assertEqualDataBoundBeans(
+            param,
+            job.getProperty(ParametersDefinitionProperty.class)
+                .getParameterDefinition("PARAM")
+        );
+    }
+
+    @Test
+    public void testConfigureParameterizedBuildFilter() throws Exception {
+        ParameterizedRunFilter filter = new ParameterizedRunFilter("${PARAM}");
+        RunSelectorStep step = new RunSelectorStep("test");
+        step.setFilter(filter);
+        new StepConfigTester(j).configRoundTrip(step);
+        j.assertEqualDataBoundBeans(filter, step.getFilter());
+    }
+
+    @Test
+    public void testIsSelectableWithDefault() throws Exception {
+        WorkflowJob selecter = j.jenkins.createProject(
+            WorkflowJob.class,
+            RandomStringUtils.randomAlphanumeric(7)
+        );
+        selecter.addProperty(new ParametersDefinitionProperty(
+            Arrays.<ParameterDefinition>asList(new RunFilterParameter(
```

Comment:
I tested `Collections.singletonList` and found that `ParameterDefinitionsProperty` provides a constractor with a variant number of parameters!
I don't need `Array.asList` nor `Collection.singletonList`.
(I used unnecessary `Array.asList` everytime I use `ParameterDefinitionsProperty` for long time...)
Thank you for let me find that.